### PR TITLE
[Model Update]: Material Demand 24.05 changes

### LINKED ISSUE #586: [Model Update]: Material Demand 24.05 changes
## Update Reason

Add `materialDemandIsInactive` flag to indicate that Material Demand is not in use by customer anymore

## MS1 Criteria

- [ ] The model that should be updated exists
- [ ] The referenced use case exists
- [ ] The potential updates are discussed with all stakeholders
- [ ] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [ ] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [ ] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by nhaenis**
## Description

Changes for DCM & PURIS standard 24.05. Changes see RELEASE_NOTES.txt

Closes #586 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.5.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [ ] All required reviewers have approved this PR (see reviewers section)
- [ ] The new aspect (version) will be implemented by at least one data provider
- [ ] The new aspect (version) will be consumed by at least one data consumer
- [ ] There exists valid test data
- [ ] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [ ] The model has at least version '1.0.0'
- [ ] If a previous model exists, model deprecation has been checked for previous model
- [ ] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.material_demand:1.0.0#> .
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.material_demand/2.0.0/MaterialDemand.ttl:34]: you can use business_partner_number 2.0.0

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+@prefix ext-quantity: <urn:samm:io.catenax.shared.quantity:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.material_demand/2.0.0/MaterialDemand.ttl:35]: quantity 2.0.0

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.material_demand/2.0.0/MaterialDemand.ttl:36]: uuid 2.0.0

Comment (nhaenis) [reply io.catenax.material_demand/2.0.0/MaterialDemand.ttl]: updated

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl (hunk 2 above)
Comment (nhaenis) [reply io.catenax.material_demand/2.0.0/MaterialDemand.ttl]: updated

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl (hunk 1 above)
Comment (nhaenis) [reply io.catenax.material_demand/2.0.0/MaterialDemand.ttl]: updated

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+   samm:description "Defines the start of the demand rate as a point in time. The point in time must be interpreted according to the demand rate."@en ;
+   samm:characteristic :PointInTimeCharacteristic ;
+   samm:exampleValue "2023-10-09"^^xsd:date .
+
+:DemandCategoryCodeCharacteristic a samm-c:Code ;
+   samm:preferredName "Demand Category Code Characteristic"@en ;
+   samm:description "The code identifying a demand category."@en ;
+   samm:dataType xsd:string .
+
+:QuantityTrait a samm-c:Trait ;
```
Comment (agg3fe) [inline io.catenax.material_demand/2.0.0/MaterialDemand.ttl:284]: preferredName and description missing

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments and then we are good for MS2

File: io.catenax.material_demand/2.0.0/MaterialDemand.ttl (hunk 4 above)
Comment (nhaenis) [reply io.catenax.material_demand/2.0.0/MaterialDemand.ttl:284]: added

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by tom-rm-meyer-ISST** [comment]
One point I found: Maybe it makes sense to update the description of the materialGlobalAssetId to be more abstract (PartAsPlanned is from now on PartTypeInformation). We used:

```
:materialGlobalAssetId a samm:Property;
    samm:preferredName "UUID of the Part Type Twin"@en;
    samm:description "Identifier used uniquely to identify part type twin."@en;
    samm:characteristic ext-uuid:UuidV4Trait;
    samm:exampleValue "urn:uuid:48878d48-6f1d-47f5-8ded-a441d0d879df".
```

**Comment by agg3fe** [comment]
> One point I found: Maybe it makes sense to update the description of the materialGlobalAssetId to be more abstract (PartAsPlanned is from now on PartTypeInformation). We used:
> 
> ```
> :materialGlobalAssetId a samm:Property;
>     samm:preferredName "UUID of the Part Type Twin"@en;
>     samm:description "Identifier used uniquely to identify part type twin."@en;
>     samm:characteristic ext-uuid:UuidV4Trait;
>     samm:exampleValue "urn:uuid:48878d48-6f1d-47f5-8ded-a441d0d879df".
> ```

@nhaenis Could you please check this.

**Comment by agg3fe** [comment]
not required.

**Pull request closed without merging**